src/mlpack/methods/cf/cf_model.hpp

```diff
@@ -0,0 +1,216 @@
+/**
+ * @file cf_model.hpp
+ * @author Wenhao Huang
+ *
+ * A serializable CF model, used by the main program.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_CF_CF_MODEL_HPP
+#define MLPACK_METHODS_CF_CF_MODEL_HPP
+
+#include <mlpack/core.hpp>
+#include "cf.hpp"
+
+
+#include <mlpack/methods/cf/decomposition_policies/batch_svd_method.hpp>
+#include <mlpack/methods/cf/decomposition_policies/randomized_svd_method.hpp>
+#include <mlpack/methods/cf/decomposition_policies/regularized_svd_method.hpp>
+#include <mlpack/methods/cf/decomposition_policies/svd_complete_method.hpp>
+#include <mlpack/methods/cf/decomposition_policies/svd_incomplete_method.hpp>
+
+namespace mlpack {
+namespace cf {
+
+/**
+ * The model to save to disk.
+ */
+class CFModel
+{
+ public:
+  enum DecompositionPolicies
+  {
+    NMF,
+    BATCH_SVD,
+    RANDOMIZED_SVD,
+    REGULARIZED_SVD,
+    SVD_COMPLETE,
+    SVD_INCOMPLETE
+  };
+
+ private:
+  //! The type of decomposition policy.
+  size_t decompositionPolicy;
+  //! Non-NULL if using NMFPolicy.
+  CFType<NMFPolicy>* nmfCF;
```

Comment:
How do you think is it a good idea to refactor this in terms of `boost::variant` like `src/mlpack/methods/neighbor_search/ns_model.hpp`?

Reply:
Yes I will work on that. Thanks!